src/Model/Post.php

```diff
@@ -88,6 +87,22 @@ public function __construct( \WP_Post $post ) {
 		$this->post = $post;
 		$this->post_type_object = isset( $post->post_type ) ? get_post_type_object( $post->post_type ) : null;
 
+		/**
+		 * Set the resolving post to the global $post. That way any filters that
+		 * might be applied when resolving fields can rely on global post and
+		 * post data being set up.
+		 */
+		$GLOBALS['post'] = $this->post;
+		setup_postdata( $this->post );
+
+		/**
+		 * Mimic core functionality for templates, as seen here:
+		 * https://github.com/WordPress/WordPress/blob/6fd8080e7ee7599b36d4528f72a8ced612130b8c/wp-includes/template-loader.php#L56
+		 */
+		if ( 'attachment' === $this->post->post_type ) {
+			remove_filter( 'the_content', 'prepend_attachment' );
+		}
```

Comment:
Should all of this stuff not happen in `DataSource`?

Reply:
No. DataSource is essentially an extension of DataLoader, and it's primary concern is asking for data, not preparing the data. 

The Model Layer's job, now, is to respond with the prepared data.

Also, because of Deferred resolution, we're not composing objects all at once and returning them all at once, we may return one field initially, then return other fields, then come back for even more fields. 

So the Model Layer pretty much _has_ to be where this happens now.